pilot/pkg/networking/core/cluster_traffic_policy.go

```diff
@@ -197,6 +197,12 @@ func shouldH2Upgrade(clusterName string, port *model.Port, mesh *meshconfig.Mesh
 	// Upgrade if tls.GetMode() == networking.TLSSettings_ISTIO_MUTUAL
 	if connectionPool != nil && connectionPool.Http != nil {
 		override := connectionPool.Http.H2UpgradePolicy
```

Comment:
nit: can you move this below the if?

Reply:
Basically this PR also fixes the case of using useClientProtocol and h2Upgrade in same DR. Istio has webhook validations but just in case, the code arrives here it will prevent that.

Reply:
ok. That seems reasonable